arch/arm/src/stm32/stm32_spi.c

```diff
@@ -1567,10 +1597,12 @@ static void spi_exchange(FAR struct spi_dev_s *dev, FAR const void *txbuffer,
   FAR struct stm32_spidev_s *priv = (FAR struct stm32_spidev_s *)dev;
 
 #ifdef CONFIG_STM32_DMACAPABLE
-  if ((txbuffer && !stm32_dmacapable((uint32_t)txbuffer, nwords, priv->txccr)) ||
-      (rxbuffer && !stm32_dmacapable((uint32_t)rxbuffer, nwords, priv->rxccr)))
+  if (!(priv->rxch && priv->txch) ||
+      (txbuffer && !stm32_dmacapable((uint32_t)txbuffer, nwords, priv->txccr)) ||
+      (rxbuffer && !stm32_dmacapable((uint32_t)rxbuffer, nwords, priv->rxccr)) ||
+      up_interrupt_context())
```

Comment:
Is up_interrupt_context() is a PX4 specific thing, but not a general usage?

Reply:
I see it all over nuttx proper, although if it seems out of place here we could do something on the PX4 side to force it. I think that will be important for finding a way to do this incrementally.

Reply:
@dagar - I think it is an artifact of calling the SPI off the HRT. We will need to change that to upstream it.

Reply:
@dagar the more i think about it this is correct. A driver using SPI in interrupt context can not do the sem_wait.